tests/src/main/java/net/neoforged/neoforge/debug/attachment/AttachmentTests.java

```diff
@@ -158,4 +167,73 @@ static void dynamicDataContentSerialization(DynamicTest test, RegistrationHelper
             helper.succeed();
         });
     }
+
+    @TestHolder(description = "Tests serialization for an empty object", enabledByDefault = true)
+    static void pseudoEmptyAttachmentSerialization(DynamicTest test, RegistrationHelper reg) {
+        class EmptyObject {
+            @Nullable
+            public Integer value;
+
+            public EmptyObject(Optional<Integer> value) {
+                this.value = value.orElse(null);
+            }
+
+            @Override
+            public boolean equals(Object obj) {
+                if (!(obj instanceof EmptyObject empty)) return false;
+                return Objects.equals(this.value, empty.value);
+            }
+        }
+
+        class HolderTest extends AttachmentHolder implements ValueIOSerializable {
+            @Override
+            public void serialize(ValueOutput output) {
+                this.serializeAttachments(output);
+            }
+
+            @Override
+            public void deserialize(ValueInput input) {
+                this.deserializeAttachments(input);
+            }
+        }
+
+        var nullableAttachment = reg.attachments()
+                .register("nullable_attachment", () -> AttachmentType.builder(() -> new EmptyObject(Optional.of(1)))
+                        .serialize(RecordCodecBuilder.mapCodec(
+                                instance -> instance.group(
+                                        Codec.INT.optionalFieldOf("value").forGetter(obj -> Optional.ofNullable(obj.value))).apply(instance, EmptyObject::new)))
+                        .build());
+
+        test.framework().modEventBus().addListener((FMLLoadCompleteEvent event) -> event.enqueueWork(() -> {
+            // Serialize and deserialize object
+            var lookup = RegistryAccess.fromRegistryOfRegistries(BuiltInRegistries.REGISTRY);
+
+            var originalHolder = new HolderTest();
+            originalHolder.setData(nullableAttachment, new EmptyObject(Optional.empty()));
+
+            var output = TagValueOutput.createWithContext(ProblemReporter.DISCARDING, lookup);
+            originalHolder.serializeAttachments(output);
+
+            var anotherHolder = new HolderTest();
+            try {
+                // Try to deserialize
+                anotherHolder.deserialize(TagValueInput.create(ProblemReporter.DISCARDING, lookup, output.buildResult()));
+            } catch (Exception e) {
```

Comment:
What's the point of this? The test will fail anyway if there is an exception.

Reply:
The test doesn't fail, but it also doesn't succeed either, leaving it as an incomplete test. The point is to purposely mark the test as failed so that the error is caught as part of the failure message rather than searching for such a log.

Reply:
I would remove the try/catch (it really should fail the test), but 🤷 this also works for me.